src/bin/languageserver/mod.rs

```diff
@@ -1158,26 +1168,52 @@ impl LanguageServer for SolangServer {
     async fn did_open(&self, params: DidOpenTextDocumentParams) {
         let uri = params.text_document.uri;
 
+        if let Ok(path) = uri.to_file_path() {
+            self.files
+                .lock()
+                .await
+                .text_buffers
+                .insert(path, params.text_document.text);
+        }
+
         self.parse_file(uri).await;
     }
 
     async fn did_change(&self, params: DidChangeTextDocumentParams) {
         let uri = params.text_document.uri;
 
+        if let Ok(path) = uri.to_file_path() {
```

Comment:
Same, I don't know this code so might be fine. But this is very unusual

Reply:
same as the previous one. Also, please feel free to let me know if you had something specific in mind that we can do when `Err` is returned in this scenario. 
Because we can't notify the client/editor about this (lack of return value) and because we don't store any state between the calls, I couldn't think of anything we could do here, other than silently ignore the request.

Reply:
Ok, probably we want to log a warning then as Sean suggested above.